xls/modules/apfloat_sub_2.x

```diff
@@ -0,0 +1,325 @@
+#![test]
+fn test_apfloat_sub_2() {
+  let one = float32::one(u1:0);
+  let two = apfloat_add_2::add<u32:8, u32:23>(one, one);
+  let neg_two = F32{sign: u1:1, ..two};
+  let three = apfloat_add_2::add<u32:8, u32:23>(one, two);
+  let four = apfloat_add_2::add<u32:8, u32:23>(two, two);
+
+  let _ = assert_eq(apfloat_sub_2(four, one), three);
+  let _ = assert_eq(apfloat_sub_2(four, two), two);
+  let _ = assert_eq(apfloat_sub_2(four, three), one);
+  let _ = assert_eq(apfloat_sub_2(three, two), one);
+  let _ = assert_eq(apfloat_sub_2(two, four), neg_two);
+  ()
+}
+
+// Returns u1:1 if x >= y.
+// Always returns false if x or y is NaN.
+pub fn apfloat_gte_2<EXP_SZ: u32, SFD_SZ: u32>(
+    x: APFloat<EXP_SZ, SFD_SZ>, 
+    y: APFloat<EXP_SZ, SFD_SZ>) -> u1 {
+  let diff = apfloat_sub_2(x, y);
+  !diff.sign if !(apfloat::is_nan(x) || apfloat::is_nan(y))
+             else u1:0
+}
+
+#![test]
+fn test_fp_gte_2() {
+  let zero = float32::zero(u1:0);
+  let one = float32::one(u1:0);
+  let neg_one = float32::one(u1:1);
+  let two = apfloat_add_2::add(one, one);
+  let neg_two = apfloat_add_2::add(neg_one, neg_one);
+  let inf = float32::inf(u1:0);
+  let neg_inf = float32::inf(u1:1);
+  let nan = float32::qnan();
+  let denormal_1 = float32::unflatten(u32:1);
+  let denormal_2 = float32::unflatten(u32:2);
+
+  // Test unequal.
+  let _ = assert_eq(apfloat_gte_2(one, two), u1:0);
+  let _ = assert_eq(apfloat_gte_2(two, one), u1:1);
+
+  // Test equal.
+  let _ = assert_eq(apfloat_gte_2(one, one), u1:1);
+  let _ = assert_eq(apfloat_gte_2(two, two), u1:1);
+  let _ = assert_eq(apfloat_gte_2(denormal_1, denormal_2), u1:1);
+  let _ = assert_eq(apfloat_gte_2(denormal_1, zero), u1:1);
+
+  // Test negatives.
+  let _ = assert_eq(apfloat_gte_2(one, neg_one), u1:1);
+  let _ = assert_eq(apfloat_gte_2(neg_one, one), u1:0);
+  let _ = assert_eq(apfloat_gte_2(neg_one, neg_one), u1:1);
+  let _ = assert_eq(apfloat_gte_2(neg_two, neg_two), u1:1);
+  let _ = assert_eq(apfloat_gte_2(neg_one, neg_two), u1:1);
+  let _ = assert_eq(apfloat_gte_2(neg_two, neg_one), u1:0);
+
+  // Special case - inf.
+  let _ = assert_eq(apfloat_gte_2(inf, neg_one), u1:1);
+  let _ = assert_eq(apfloat_gte_2(neg_inf, inf), u1:0);
+  let _ = assert_eq(apfloat_gte_2(inf, inf), u1:1);
+  let _ = assert_eq(apfloat_gte_2(inf, two), u1:1);
+  let _ = assert_eq(apfloat_gte_2(neg_two, neg_inf), u1:1);
+
+  // Special case - NaN (always returns false).
+  let _ = assert_eq(apfloat_gte_2(one, nan), u1:0);
+  let _ = assert_eq(apfloat_gte_2(nan, one), u1:0);
+  let _ = assert_eq(apfloat_gte_2(neg_one, nan), u1:0);
+  let _ = assert_eq(apfloat_gte_2(nan, neg_one), u1:0);
+  let _ = assert_eq(apfloat_gte_2(inf, nan), u1:0);
+  let _ = assert_eq(apfloat_gte_2(nan, inf), u1:0);
+  let _ = assert_eq(apfloat_gte_2(nan, nan), u1:0);
+
+  ()
+}
+
+// Returns u1:1 if x == y.
+// Always returns false if x or y is NaN.
+pub fn apfloat_eq_2<EXP_SZ: u32, SFD_SZ: u32>(
```

Comment:
Should this live somewhere else? apfloat.x or some apfloat comparison library. Same below for other comparisons.

@RobSpringer any comment?

Reply:
Done.